src/Nethermind/Nethermind.Facade/Simulate/SimulateReadOnlyBlocksProcessingEnvFactory.cs

```diff
@@ -48,6 +48,7 @@ public ISimulateReadOnlyBlocksProcessingEnv Create()
             .AddDecorator<IBlockhashProvider, SimulateBlockhashProvider>()
             .AddDecorator<IVirtualMachine, SimulateVirtualMachine>()
             .AddDecorator<IBlockValidator, SimulateBlockValidatorProxy>()
+            .AddSingleton<ITransactionProcessor, SimulateTransactionProcessor>()
```

Comment:
Got a solution that does not involve a subclass of `SimulateTransactionProcessor`? This will break plugins that override `ITransactionProcessor`. Perhaps some kind gas calculator interface  maybe?

Reply:
I would move all virtual functions into an interface, but it's overkill probably